src/Psalm/Internal/Analyzer/Statements/Expression/Fetch/ClassConstFetchAnalyzer.php

```diff
@@ -452,6 +456,255 @@ public static function analyze(
 
         $context->inside_general_use = $was_inside_general_use;
 
+        if ($stmt->class instanceof PhpParser\Node\Expr\Variable) {
+            $type = $statements_analyzer->node_data->getType($stmt->class);
+
+            $fq_class_name = null;
+            if ($type !== null && $type->isSingle()) {
+                $atomic_type = \array_values($type->getAtomicTypes())[0];
+                if ($atomic_type instanceof TNamedObject) {
+                    $fq_class_name = $atomic_type->value;
+                }
```

Comment:
We could add support for class-strings here, to be able to infer more type, e.g.: https://3v4l.org/9Kdhd